Bio/SeqUtils/MeltingTemp.py

```diff
@@ -468,7 +473,9 @@ def _check(seq, method):
         )
     if method == "Tm_NN":
         baseset = ("A", "C", "G", "T", "I")
-    seq = "".join([base for base in seq if base in baseset])
+
+    if not all(base in baseset for base in seq):
```

Comment:
I'm not sure how computationally intense this module is, but this is a fairly inefficient operation. Tm_Wallace went from about 4s per million operations to 15s with checks online compared to previous main on a 140 bp sequence. We can ameliorate that down to 6-7s by switching the basesets to sets and this line to:
```
if not set(seq).issubset(baseset):
```
The remaining extra time is mostly coming from the isinstance checks and re.sub, but that's probably fine because you can turn off checks and we're getting more robustness (hopefully). Certainly it could be improved on further, but the low hanging fruit are probably fine unless this computation is a bottleneck for anyone

Reply:
The comparison with the actual `Tm_Wallace` is not really fair, because the sequence was not checked for valid characters, but just returned. A comparison either with `Tm_NN` or `Tm_GC` would be more fair. 

Nevertheless, I would also go for a set-based comparison here, as you (@flaar94) suggests.